youtube_dl/extractor/iqiyi.py

```diff
@@ -302,7 +302,7 @@ def _extract_playlist(self, webpage):
         PAGE_SIZE = 50
 
         links = re.findall(
-            r'<a[^>]+class="site-piclist_pic_link"[^>]+href="(http://www\.iqiyi\.com/.+\.html)"',
+            r'<a[^>]+?href="(http://www\.iqiyi\.com/.+\.html)"[^>]+?class="site-piclist_pic_link"[^>]+?>',
```

Comment:
@dstftw 
I removed the ? - I thought maybe lazy matching would be better.
This regex now should capture the URL in all cases where 'href' ist before 'sit-piclist_pic_link'. If it's the other way around (making "href" not the first <a>-Attribute is valid but seldom used) it would fail.

I suppose I can make the regex so that the other way around is also matched - but this would make it more complex - do you think thats a good idea?